cmd/submit.go

```diff
@@ -114,6 +121,23 @@ func runSubmit(cfg config.Config, flags *pflag.FlagSet, args []string) error {
 	return nil
 }
 
+func getExerciseSolutionFiles(baseDir string) ([]string, error) {
+	v := viper.New()
+	v.AddConfigPath(filepath.Join(baseDir, ".exercism"))
+	v.SetConfigName("config")
+	v.SetConfigType("json")
+	err := v.ReadInConfig()
+	if err != nil {
+		return nil, errors.New("no files to submit")
+	}
+	solutionFiles := v.GetStringSlice("files.solution")
+	if len(solutionFiles) == 0 {
+		return nil, errors.New("no files to submit")
+	}
```

Comment:
Haven't written any go in a while, but I think that this is all we'd need to support globs nicely (depending on https://github.com/yargevad/filepathx):

```suggestion
	patterns := v.GetStringSlice("files.solution")
	if len(patterns) == 0 {
		return nil, errors.New("no files to submit")
	}
	
	solutionFiles := make([]String, len(patterns))
	for _, pattern := range patterns {
	    matches, err := filepathx.Glob(pattern)
	    if err != nil {
	        return nil, errors.New("invalid glob pattern") // TODO: include bad pattern in err
	    }
	    
	    solutionFiles = append(solutionFiles, matches...)
	}
```

In the event that the patterns are path literals, then this keeps current behavior. Otherwise, this gets all files which match any glob pattern in the list.

I'd personally prefer to see this included in this PR instead of needing to create a new issue and PR.

Reply:
It'd would be something like that, yes, but the logic would have to be inside the `runSubmit` function and not here. This function only runs when no paths were given and I see no point on trying to resolve globs from the paths from the `files.solution` key as I don't think it's even possible to have globs there (we don't have them in the Go track and quickly checking some exercises on the Rust track, I can't see exercises using globs in their `files.solution`).

I don't mind implementing globs and the support for specifying just a directory for upload. I'm kinda just waiting on greenlight for those features, because they weren't planned or discussed properly.

Reply:
It's a chicken and egg problem. Right now, globs there are not supported, so we're not using them. Adding support in this tool isn't sufficient to call them Supported by Exercism, but it's necessary. 

What kind of discussion did you have in mind, to qualify as planned or discussed properly?

Reply:
> What kind of discussion did you have in mind, to qualify as planned or discussed properly?

Having more than one person weight on it at least? In particular I'd like the opinion of someone more familiar with the CLI.

I mean, this is my first PR for the CLI, I don't even know if the way I did the "no files" thing is the right way to do it given the current codebase and we're already talking about adding more features on top of it.

I'm not saying that these aren't features I would like to see, I like them, just don't see why we can't do it in baby steps and have more feedback on each of them.

Reply:
Yes, it's only necessary if the tool and exercism must support globs. I believe that to be the case, but I also believe that merging some kind of submit-without-specifying is more important than the specific globs feature.

I don't intend to be a drag on the process here. I'm just convinced that:

- in several tracks, students can split up their work into modules which live in different files with arbitrary names
- splitting work into several files is often the right thing to do, and we should encourage it in those cases
- auto-submit without globs will always fail for a student who has done that

This implies that globs are a feature we'll want to support eventually.

Reply:
I can definitely see the arguments in favor of globs, so let's discuss them in a separate issue.

Reply:
Made the separate issue: https://github.com/exercism/exercism/issues/6399 . It's in `/exercism` instead of `/cli` because I'm pretty sure that support for that feature will require changes in at least two repos.

Reply:
Great, thanks!